index.js

```diff
@@ -339,24 +292,145 @@ module.exports = {
   TYPE_REJECT,
   TYPE_MESSAGE,
 
+  typeToString,
+
   MIME_APPLICATION_OCTET_STREAM,
   MIME_TEXT_PLAIN_UTF8,
   MIME_APPLICATION_JSON,
 
-  serializeAck,
-  serializeResponse,
-  serializeError,
-  serializePrepare,
-  serializeFulfill,
-  serializeReject,
-  serializeMessage,
-
-  deserializeAck,
-  deserializeResponse,
-  deserializeError,
-  deserializePrepare,
-  deserializeFulfill,
-  deserializeReject,
-  deserializeMessage,
-  deserializeClpPacket
+  serialize,
+  deserialize,
+
+  // The following legacy functions use an alternative format to access the exposed
+  // serialize/deserialize functionality. There is one pair of such functions per CLP call.
+  // The object returned by the deserialize function differs per call, is different from
+  // the arguments passed to the corresponding serialize function, and is also different
+  // from the structure of CLP's OER encoding. Due to these disadvantages, these functions
+  // are marked as 'legacy functions' here; however, we did not remove them because
+  // the serialize ones are already used by the payment plugin framework.
+
+  serializeAck (requestId, protocolData) {
+    return serialize({
+      type: TYPE_ACK,
+      requestId,
+      data: protocolData
+    })
+  },
+  deserializeAck (buffer) {
+    const obj = deserialize(buffer)
+    return {
+      requestId: obj.requestId,
```

Comment:
There is an inconsistency between `deserialize` and the deserialize_Type_() functions. The former returns `requestId` as a field in the envelope, the latter returns `requestId` as a field in data.

So if you compare the objects returned from `clp.deserialize(packet).data` to clp.deserializePrepare(packet)` the objects will not be the same because of `requestId`.

Reply:
How about putting `requestId` always into the `data` field?

Reply:
Yes, `clp.deserialize` follows the OER format, and `clp.deserializePrepare` etc follow the slightly different format which Ben introduced. The reason these legacy functions are still there is that they are used in the payment channel framework. I implemented them because you said ['Removing the serialize* functions breaks a lot of existing code'](https://github.com/interledgerjs/clp-packet/pull/4#pullrequestreview-58938324) in your previous review. Changing them would break that existing code just as much as removing them.

Reply:
Michiel and I settled over slack to remove the type specific deserialization functions (`clp.deserializePrepare(..)` etc.).